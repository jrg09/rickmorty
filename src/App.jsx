import { useState } from "react";
import { Header } from "./components/Header";
import { List } from "./components/List";

export const App = () => {
  return (
    <>
      <div className="container p-2">
        <Header />
        <div className="p-2">
          <p>Este proyecto hace uso de:</p>
          <ul>
            <li>
              API de Rick and Morty (
              <a
                href="https://rickandmortyapi.com/api/character"
                target="_blank"
              >
                https://rickandmortyapi.com/api/character
              </a>
              )
            </li>
            <li>
              Bootstrap y&nbsp;
              <a href="https://react-bootstrap.github.io/getting-started/introduction">
                React-Bootstrap
              </a>
            </li>
            <li>Componentes separados (Header, List y Character)</li>
          </ul>
          <List />
        </div>
      </div>
    </>
  );
};
