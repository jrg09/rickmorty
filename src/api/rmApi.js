export const getRickMortyCharactersAPI = () => {
  return fetch("https://rickandmortyapi.com/api/character").then((resp) => {
    if (resp.status === 200) return resp.json();
    else throw new Error("Call to RickAndMorty API was wrong");
  });
};
