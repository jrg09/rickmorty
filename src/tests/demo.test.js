import { render, screen, waitFor } from "@testing-library/react";
import { getRickMortyCharactersAPI } from "../api/rmApi";
import { App } from "../App";

describe("datosDescribe", () => {
  test("should hacer match con snapshot", () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });

  test("should call API ok with more than 1 character and contains name and origin", async () => {
    const { results } = await getRickMortyCharactersAPI();

    expect(results.length).toBeGreaterThan(0);

    expect(results[0].name).toBeTruthy();
    expect(results[0].origin).toBeTruthy();
  });

  test("should contener elementos de tipo Card", async () => {
    const { container } = render(<App />);

    await waitFor(() => {
      screen.getByText("Rick Sanchez");
      //screen.debug();
      const ds = container.querySelectorAll(".card");
      expect(ds.length).toBeGreaterThan(0);
    });
  });
});
