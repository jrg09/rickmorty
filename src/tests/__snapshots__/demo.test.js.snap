// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`datosDescribe should hacer match con snapshot 1`] = `
<div>
  <div
    class="container p-2"
  >
    <div
      class="navbar sticky-top navbar navbar-light bg-dark nav"
    >
      <h1
        class="navbar-brand text-light p-2"
      >
        Rick and Morty API
      </h1>
    </div>
    <div
      class="p-2"
    >
      <p>
        Este proyecto hace uso de:
      </p>
      <ul>
        <li>
          API de Rick and Morty (
          <a
            href="https://rickandmortyapi.com/api/character"
            target="_blank"
          >
            https://rickandmortyapi.com/api/character
          </a>
          )
        </li>
        <li>
          Bootstrap yÂ 
          <a
            href="https://react-bootstrap.github.io/getting-started/introduction"
          >
            React-Bootstrap
          </a>
        </li>
        <li>
          Componentes separados (Header, List y Character)
        </li>
      </ul>
      <div
        class="row mt-3"
      >
        <div>
          Cargando...
        </div>
      </div>
    </div>
  </div>
</div>
`;
